staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -382,6 +384,9 @@ loop:
 			if event.Type == watch.Error {
 				return apierrs.FromObject(event.Object)
 			}
+			if event.Type == watch.HeartBeat {
```

Comment:
how would old clients handle this watch event?

Reply:
The behavior now is when receiving unknown watch event type, it will break the loop and re-list.

Reply:
How about disabling heartbeat by default?

Reply:
I think clients would have to specifically request heartbeats for this to work.

Reply:
@lavalamp yes,  one way is to add a field `KeepAliveTime` to ListOptions,  which can be set when doing watch request, just like `TimeoutSeconds`.